dashboards/resources/node.libsonnet

```diff
@@ -1,154 +1,308 @@
+        node:
+          var.query.new('node')
+          + var.query.withDatasourceFromVariable(self.datasource)
+          + var.query.queryTypes.withLabelValues(
+            'node',
+            'kube_node_info{%(clusterLabel)s="$cluster"}' % $._config
+          )
+          + var.query.generalOptions.withLabel('node')
+          + var.query.refresh.onTime()
+          + var.query.generalOptions.showOnDashboard.withLabelAndValue()
+          + var.query.selectionOptions.withMulti(true),
+      };
+
+      local links = {
         pod: {
-          alias: 'Pod',
+          title: 'Drill down to pods',
+          url: '%(prefix)s/d/%(uid)s/k8s-resources-pod?${datasource:queryparam}&var-cluster=$cluster&var-namespace=$namespace&var-pod=${__data.fields.Pod}' % {
+            uid: $._config.grafanaDashboardIDs['k8s-resources-pod.json'],
+            prefix: $._config.grafanaK8s.linkPrefix,
+          },
         },
       };
 
-      g.dashboard(
-        '%(dashboardNamePrefix)sCompute Resources / Node (Pods)' % $._config.grafanaK8s,
-        uid=($._config.grafanaDashboardIDs['k8s-resources-node.json']),
-        datasource_regex=$._config.datasourceFilterRegex,
-        datasource=$._config.datasourceName,
-      )
-      .addRow(
-        g.row('CPU Usage')
-        .addPanel(
-          g.panel('CPU Usage') +
-          g.queryPanel([
+      local panels = [
+        tsPanel.new('CPU Usage')
+        + tsPanel.queryOptions.withTargets([
+          prometheus.new(
+            '${datasource}',
             'sum(kube_node_status_capacity{%(clusterLabel)s="$cluster", node=~"$node", resource="cpu"})' % $._config,
+          )
+          + prometheus.withLegendFormat('max capacity'),
+
+          prometheus.new(
+            '${datasource}',
             'sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate{%(clusterLabel)s="$cluster", node=~"$node"}) by (pod)' % $._config,
-          ], [
-            'max capacity',
-            '{{pod}}',
-          ]) +
-          g.stack +
+          )
+          + prometheus.withLegendFormat('{{pod}}'),
+        ])
+        + tsPanel.fieldConfig.defaults.custom.withStacking({ mode: 'normal' })
+        + tsPanel.standardOptions.withOverrides([
+          fieldOverride.byName.new('max capacity')
+          + fieldOverride.byName.withPropertiesFromOptions(
+            timeSeries.standardOptions.color.withMode('fixed')
+            + timeSeries.standardOptions.color.withFixedColor('red')
+          )
+          + fieldOverride.byName.withProperty('custom.stacking', { mode: 'none' })
+          // This effectively "hides" max capacity from the panel. It shows only in the legend, showing the max capacity. In the "legacy" graph panel
+          // the max capacity could be reselected and thus shown on the timeseries with a dotted line. This is no longer possible.
+          + fieldOverride.byName.withProperty('custom.hideFrom', { tooltip: true, viz: true, legend: false })
```

Comment:
Seeking feedback on this from @skl and maintainers.

For many of the timeseries panels which show requests and limits as visible dotted line thresholds on the panel, the "capacity" on this dashboard, and the compute/namespace (pod) are only shown in the legend, and can not be "unhid" to show on the timeseries.

Is this desirable? Should the capacity be shown in the graph?

In under utilized environments (like mine) the capacity is dramatically higher than the stacked usage, so you would want to hide it to be able to reason on the actual utilization.

Reply:
I'm ok with just seeing the max cpu/memory capacity of the node in the legend like this, especially as that value probably changes rarely.

For k8s requests/limits/quotas though, I think it's useful to see them plotted as series on the graph as these can change much more frequently.

@povilasv be good to get your opinion here, too.